src/nbsap/views/goals.py

```diff
@@ -2,33 +2,147 @@
 from cStringIO import StringIO
 
 from django.shortcuts import render, get_object_or_404
-from django.core.paginator import Paginator
 from django.conf import settings
-from django.http import HttpResponse
+from django.http import HttpResponse, Http404
 from django.template import RequestContext
 from django.template.loader import render_to_string
 from django.shortcuts import render_to_response
 
 from nbsap import models
-from indicators import get_indicators_pages
+from nbsap.models import sort_by_code
 
 
-def goals(request, code):
-    current_goal = get_object_or_404(models.AichiGoal, code=code)
+def get_most_relevant_targets(target):
+    most_relevant_targets = []
+    if target.eu_aichi_strategy.count():
+        for strategy in target.eu_aichi_strategy.all():
+            for eu_target in strategy.eu_targets.all():
+                most_relevant_targets.append(eu_target)
+    return sort_by_code(most_relevant_targets)
+
+
+def get_other_relevant_targets(target):
+    other_relevant_targets = []
+    if target.eu_other_aichi_strategy.count():
+        for strategy in target.eu_other_aichi_strategy.all():
+            for eu_target in strategy.eu_targets.all():
+                other_relevant_targets.append(eu_target)
+    return sort_by_code(other_relevant_targets)
+
+
+def get_most_relevant_indicators(target):
+    most_relevant_indicators = []
+    if target.eu_indicator_aichi_strategy.count():
+        for strategy in target.eu_indicator_aichi_strategy.all():
+            most_relevant_indicators.append(strategy.eu_indicator)
+    return most_relevant_indicators
+
+
+def get_adjacent_targets(targets, current_target):
+    if not current_target:
+        return None, None
+
+    for index, target in enumerate(targets):
+        if target == current_target:
+            previous_index = index - 1
+            next_index = index + 1
+            break
+
+    if previous_index < 0:
+        previous_index = len(targets) - 1
+    if next_index > len(targets) - 1:
+        next_index = 0
+
+    return targets[previous_index], targets[next_index]
+
+
+def user_homepage(request):
+    return render(request, 'user_homepage.html')
+
+
+def list_goals(request):
     goals = models.AichiGoal.objects.order_by('code').all()
-    indicators_list = models.AichiIndicator.objects.all()
+    list_goals = True
+    list_targets = False
+    return render_to_response(
+        'aichi/aichi.html',
+        context_instance=RequestContext(
+            request, {
+                'goals': goals,
+                'list_goals': list_goals,
+                'list_targets': list_targets,
```

Comment:
You could use the values `True` and `False`, without storing them into variables.